src/lib/y2firewall/widgets/allowed_services.rb

```diff
@@ -0,0 +1,170 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2firewall/firewalld"
+require "y2firewall/widgets/services_table"
+
+module Y2Firewall
+  module Widgets
+    # This class implements a widget which allows the user to select which services
+    # should be allowed for a given zone.
+    class AllowedServices < ::CWM::CustomWidget
+      # Constructor
+      #
+      # @param zone [Y2Firewall::Firewalld::Zone] Zone
+      def initialize(zone)
+        textdomain "firewall"
+        @zone = zone
+        self.widget_id = "allowed_services"
+        @known_svcs_table = ServicesTable.new(widget_id: "known:#{zone.name}")
```

Comment:
`known_svcs_table` and `allowed_svcs_table` look like the more error-prone abbreviations ever. I can hear the typos coming. :-)